lib/mongoose_push/api/v1.ex

```diff
@@ -0,0 +1,41 @@
+defmodule MongoosePush.API.V1 do
+  @moduledoc false
+
+  use Maru.Router
+  version "v1"
+
+  plug Plug.Parsers,
+      pass: ["application/json", "text/json"],
+      json_decoder: Poison,
+      parsers: [:urlencoded, :json, :multipart]
+
+  params do
+    requires  :service,       type: Atom, values: [:fcm, :apns]
+    requires  :body,          type: String
+    requires  :title,         type: String
+    optional  :badge,         type: Integer
+    optional  :click_action,  type: String
+    optional  :tag,           type: String
+    optional  :topic,         type: String
+    optional  :mode,          type: Atom, values: [:prod, :dev]
+  end
+
+  namespace :notification do
+    route_param :device_id do
+      post do
+        device_id = params.device_id
+        case MongoosePush.push(device_id, Map.delete(params, :device_id)) do
+          :ok ->
+            conn
+            |> put_status(200)
+            |> json(nil)
+          {:error, reason} ->
+            conn
+            |> put_status(500)
+            |> json(%{:details => inspect reason})
```

Comment:
The module is beautiful. Simple and readable. One issue though, the `inspect` from the `Kernel` module is not to be used to format data that you return to the client. You have to format you data differently, without inspect.